world-server/src/world/chat/ChatHandlers.ts

```diff
@@ -0,0 +1,46 @@
+import { Messages, Router, Logger } from 'world-core';
+import { BroadcastMessage } from '../../network';
+import { ChatMessage } from './ChatMessages';
+import { Player } from '../player';
+
+export class ChatHandler implements Messages.MsgHandler {
+    private players: Player[];
+
+    constructor(players: Player[]) {
+        this.players = players;
+    }
+
+    public getTypes(): string[] {
+        return ['chat.say'];
+    }
+
+    public validate(msg: Messages.Message): boolean {
+        let message = msg as ChatMessage;
+        if (message == null || message == undefined) {
+            Logger.warn(`Received player message does not extend ChatMessage`);
+            return false;
+        }
+        if (message.text == null || message.text == undefined) {
+            return false;
+        }
+
+        if (message.text.length == 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    public handle(msg: Messages.Message): void {
+        let message = msg as ChatMessage;
+        switch (message.type) {
+            case 'chat.say':
+                let chatMessage = message as ChatMessage;
+                chatMessage.id = msg.clientId;
```

Comment:
Shouldn't we assign a unique id to chat messages?

Reply:
Added